atc/metric/emitter/newrelic.go

```diff
@@ -20,22 +23,29 @@ type (
 	}
 
 	NewRelicEmitter struct {
-		client     *http.Client
-		url        string
-		apikey     string
-		prefix     string
-		containers *stats
-		volumes    *stats
+		Client              *http.Client
+		Url                 string
+		apikey              string
+		prefix              string
+		containers          *stats
+		volumes             *stats
+		BatchSize           int
+		BatchDuration       time.Duration
+		CompressionDisabled bool
+		LastEmitTime        time.Time
+		NewRelicBatch       []NewRelicEvent
 	}
 
 	NewRelicConfig struct {
-		AccountID     string `long:"newrelic-account-id" description:"New Relic Account ID"`
-		APIKey        string `long:"newrelic-api-key" description:"New Relic Insights API Key"`
-		ServicePrefix string `long:"newrelic-service-prefix" default:"" description:"An optional prefix for emitted New Relic events"`
+		AccountID           string        `long:"newrelic-account-id" description:"New Relic Account ID"`
+		APIKey              string        `long:"newrelic-api-key" description:"New Relic Insights API Key"`
+		ServicePrefix       string        `long:"newrelic-service-prefix" default:"" description:"An optional prefix for emitted New Relic events"`
+		BatchSize           uint64        `long:"newrelic-batch-size" default:"2000" description:"Number of events to batch together before emitting"`
+		BatchDuration       time.Duration `long:"newrelic-batch-duration" default:"60s" description:"Length of time to wait between emitting until all currently batched events are emitted"`
+		CompressionDisabled bool          `long:"newrelic-batch-compression-disabled" description:"Compress the batch before emitting"`
```

Comment:
This was originally `CompressionEnabled`. The problem we ran into that resulted in the name `CompressionDisabled` is that we want compression on by default but you can't specify `default:"true"` with the `go-flags` library, as you correctly guessed. `go-flags` throws an error during compilation. We wanted to avoid having any checks around whether `CompressionEnabled` has been set or not; it just gets tricky to think about and you get those weird `!= true/false` type of `if` checks that are just annoying to reason about.

If we had access to this: https://godoc.org/github.com/jessevdk/go-flags#Option.IsSet
then we could do `CompressionEnabled` nicely. It's not clear to me how to pull this `Option` type in though.

I'm not happy that we settled on `CompressionDisabled` but that's what motivated us to stick with it.